ledger-service/utils/src/main/scala/com/digitalasset/ledger/service/LedgerReader.scala

```diff
@@ -22,23 +22,44 @@ import Scalaz._
 
 object LedgerReader {
 
+  // TODO(Leo): need something stronger
+  type Error = String
```

Comment:
I am not sure this is needed. Using `String` in the left side of either as error message is a general pattern in our code base. We have more than 500 `Either[String, X]` or `String \/ X`